packages/components/file-uploader/src/file-uploader.js

```diff
@@ -0,0 +1,178 @@
+
+import { uploadFile } from './utils/upload-file.js';
+
+customElements.define(
+	'ts-file-uploader',
+	class extends TSElement {
+		constructor() {
+			super();
+			this.state = 'upload';
+			this.size = 'full';
+			this.files = [];
+			this.multiple = false;
+			this.dragCounter = 0;
+			this.handleUpload = this.handleUpload.bind(this);
+		}
+
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				state: { type: String, reflect: true },
+				endpoint: { type: String, reflect: true },
+				acceptedFileExtensions: {
+					type: Array,
+					reflect: true,
+					attribute: 'accepted-file-extensions'
+				},
+				rtl: { type: Boolean, reflect: true },
+				maxFileSize: {
+					type: Number,
+					reflect: true,
+					attribute: 'max-file-size'
+				},
+				files: { type: Array, reflect: true },
+				disabled: { type: Boolean, reflect: true },
+				multiple: { type: Boolean, reflect: true },
+				size: { type: String, reflect: true }
+			};
+		}
+
+		get maxFileSizeInBytes() {
+			return this.maxFileSize * 1024;
+		}
+
+		get fileCards() {
+			return this.files.map((file, index) => {
+				return html`
+					<ts-file-card
+						file-object="${JSON.stringify(file)}"
+						state="${file.state}"
+						size="${this.size}"
+						accepted-file-extensions="${JSON.stringify(
+							this.acceptedFileExtensions
+						)}"
+						error-message="${this.getErrorMessages(
+							file.isFileSizeValid,
+							file.isFileTypeValid,
+							file.errorMessage
+						)}"
+						?data-rtl="${this.rtl}"
+						@remove="${e => {
+							// eslint-disable-line lit/no-template-bind
+							this.removeFile(e, index);
+						}}"
+						@download="${e => {
+							// eslint-disable-line lit/no-template-bind
+							this.downloadFile(e, index);
+						}}"
+					></ts-file-card>
+				`;
+			});
+		}
+
+		render() {
+			return html`
+				${this.state === 'upload' || this.multiple
+					? html`
+							<ts-file-uploader-input
+								size="${this.size}"
+								?rtl="${this.rtl}"
+								?disabled="${this.disabled}"
+								?multiple="${this.multiple}"
+								accepted-file-extensions="${JSON.stringify(
+									this.acceptedFileExtensions
+								)}"
+								max-file-size="${this.maxFileSizeInBytes}"
+								@change="${this.handleUpload}"
+							></ts-file-uploader-input>
+					  `
+					: ''}
+				${this.files ? this.fileCards : ''}
+			`;
+		}
+
+		removeFile(e, index) {}
+
+		handleUpload(e) {
+			this.state = 'uploading';
+
+			if (e.detail.files) {
+				for (let i = 0; i < e.detail.files.length; i++) {
+					const file = e.detail.files[i];
+					const index = this.files.length;
+
+					const isFileSizeValid = this.isFileSizeValid(file.size);
+					const isFileTypeValid = this.isFileTypeValid(
+						file.name.split('.').pop()
+					);
+					const isValid = isFileSizeValid && isFileTypeValid;
+					console.log('const isFileTypeValid', isFileTypeValid);
```

Comment:
Yeah, this PR is still needs changes because we didn't decided wether we are going to have a component like this or not! I close it for now, if we decided to have something like this I'll create another one! 👍